include/DataflowPass.h

```diff
@@ -14,11 +14,24 @@
 #include "RunAnalysis.h"
 #include "Utils.h"
 
+// the dataflow pass is an abstract class that handles the program flow and
+// enforces the application of transfer functions and least upper
+// bound (lub) operations, provided by from subtypes of this class)
+// only specific details on how to lub and how
+// to update a set of methods given the introduction of a new method being called.
+// this class is extended by the `MustCall` pass and `CalledMethods` pass.
 class DataflowPass {
   private:
+    // looks at the cfg to analyze the flow between
+    // the branches (nodes of the graph) and performs lub between
+    // each branch and their successors.
```

Comment:
Here and elsewhere, it's unclear to me whether the `CFG` type represents a complete control flow graph or a single node in the logical CFG.